src/test/index.js

```diff
@@ -1,27 +1,481 @@
-describe( 'Test Jetpack blocks', () => {
-	it( 'should setup the editor for jetpack without failing', () => {
-		const mockRegisterBlockCollection = jest.fn();
-		jest.mock( '@wordpress/blocks', () => {
-			return {
-				getCategories: () => [ { slug: 'media' } ],
-				setCategories: jest.fn(),
-				registerBlockCollection: mockRegisterBlockCollection,
-				withBlockContentContext: jest.fn(),
+import {
+	getBlockTypes,
+	registerBlockType,
+	unregisterBlockType,
+} from '@wordpress/blocks';
+import { select } from '@wordpress/data';
+import { defaultHooks } from '@wordpress/hooks';
+import { JETPACK_DATA_PATH } from '../../jetpack/projects/plugins/jetpack/extensions/shared/get-jetpack-data';
+
+/**
+ * These are integration tests written with the intention of asserting the initialization of the Gutenberg Mobile
+ * editor. As they are integration tests, modules that were being mocked globally are now unmocked to use their
+ * real implementations.
+ *
+ * The intention of these integration tests is to ensure that:
+ * 0. Jetpack initialization is handled correctly by registering the Jetpack Block Collection, initializing Jetpack Data,
+ *    and registering all Jetpack Block Types that you may be capable of using.
+ * 1. Allowed block types are displayed correctly in the block selector if a subset of blocks is provided to be shown
+ *    or hidden.
+ *
+ * The tests are staged such that `require( '../index' )` mimics the entry point of the initialization of the Gutenberg
+ * Mobile editor, and `defaultHooks.doAction( 'native.render', props )` is the `native.render` hook that would normally be
+ * listened for when the editor is initializing. So all the tests ensure that when certain `props` are passed in from the
+ * `native.render` hook that the underlying Redux Store state is reduced expectedly. This is deliberately done to keep
+ * the tests as black-box as possible such that the implementation details of the application code may be changed more
+ * freely without a tight coupling on the code that tests it.
+ */
+describe( 'Gutenberg Mobile Initialization', () => {
+	beforeAll( () => {
+		jest.unmock( '@wordpress/api-fetch' );
+		jest.unmock( '@wordpress/blocks' );
+	} );
+
+	describe( 'Jetpack Blocks', () => {
+		it( `should register Jetpack block collection.`, () => {
+			// Arrange
+			const expectedCollectionTitle = 'Jetpack';
+			const props = { jetpackState: { isJetpackActive: false } };
+			require( '../index' );
```

Comment:
I haven't tested it but would it be possible to import this only once as the rest of the import statements, located at the top of the file? I don't see anything on [that file](https://github.com/wordpress-mobile/gutenberg-mobile/blob/a6fe7103497041bd37902d4415786618d4678c63/src/index.js) that would require to import it on every test 🤔, am I right?